drivers/hwinfo/hwinfo_stm32.c

```diff
@@ -113,6 +114,28 @@ int z_impl_hwinfo_get_reset_cause(uint32_t *cause)
 	}
 #endif
 
+#if defined(CONFIG_SOC_SERIES_STM32H7X) && defined(CORE_CM4)
+	if (LL_PWR_CPU2_IsActiveFlag_SB()) {
+		flags |= RESET_LOW_POWER_WAKE;
+	}
+#elif defined(CONFIG_SOC_SERIES_STM32H7X) && defined(CORE_CM7)
+	if (LL_PWR_CPU_IsActiveFlag_SB()) {
+		flags |= RESET_LOW_POWER_WAKE;
+	}
+#elif defined(CONFIG_SOC_SERIES_STM32MP1X)
+	if (LL_PWR_MCU_IsActiveFlag_SB()) {
+		flags |= RESET_LOW_POWER_WAKE;
+	}
+#elif defined(CONFIG_SOC_SERIES_STM32WLX) || defined(CONFIG_SOC_SERIES_STM32WBX)
+	if (LL_PWR_IsActiveFlag_C1SB()) {
+		flags |= RESET_LOW_POWER_WAKE;
+	}
```

Comment:
Could you check for the second core with CORE_CM0PLUS on STM32WL and STM32WB families, like you do for STM32H7X? I see preprocessor conditionals for CORE_CM0PLUS in the STM32WL HAL, but not in STM32WB so I'm not sure if it's possible there.

Not even sure Zephyr is supported or has been tested on those M0+ cores, but I think it'd be good practice to extend the functionality added in this PR to cover those possible use cases as well.

Reply:
Ditto for STM32MP1, as Zephyr supports Cortex-A7 processors, though you'd also need to consider the dual core A7 with a M4 MCU variants.

Reply:
> Could you check for the second core with `CORE_CM0PLUS` on STM32WL and STM32WB families, like you do for STM32H7X? I see preprocessor conditionals for `CORE_CM0PLUS` in the STM32WL HAL, but not in STM32WB so I'm not sure if it's available there.
> Not even sure Zephyr is supported or has been tested on those M0+ cores, but I think it'd be good practice to extend the functionality added in this PR to cover those possible use cases as well.

Disagree. Those are not supported atm, and maybe never will be. Let's keep things simple.
If ever added, this will have to be considered for sure, such as impact on other drivers.

Reply:
> Ditto for STM32MP1, as Zephyr supports Cortex-A7 processors, though you'd also need to consider the dual core A7 with a M4 MCU variants.

Ditto. Not supported hence not to be considered.

Reply:
Fair enough, good points about those additions being untestable without working support for multi-core Zephyr on those platforms.

Otherwise in my limited knowledge of the STM32 HALs this PR looks good to me :+1: